draft-ounsworth-cfrg-kem-combiners.mkd

```diff
@@ -280,13 +318,16 @@ This is a non-comprehensive list, further information can be found in paragraph
 
 # Practical instantiations {#sec-instantiation}
 
-The KDF MUST be instantiated with one of the following Keccak-based option.
-Each instance defines a function to be used as `KDF`, a hash `H` function to optionally derive the `k_i`, and a `counter`.
+The KDF must be instantiated with cryptographically-secure choices for `KDF` and `H`. The following are RECOMMENDED Keccak-based instatiations, but other choices MAY be made depending on availability of cryptographic primitives, protocol design needs, and especially to allow for future cryptographic agility.
+
+Each instance defines a function to be used as `KDF`, a hash `H` function to derive the `k_i`, and optionally a `counter`.
 
 1. `KDF = SHA3-256` and `H = SHA3-256`, with `hashSize = 256 bit`.
-2. `KDF = SHA3-512` and `H = SHA3-512`, with `hashSize = 512 bit`.
-3. `KDF = KMAC128` and `H = SHA3-256`, with `hashSize = 128 bit`.
-4. `KDF = KMAC256` and `H = SHA3-512`, with `hashSize = 256 bit`.
+1. `KDF = SHA3-512` and `H = SHA3-512`, with `hashSize = 512 bit`.
+1. `KDF = SHAKE128` and `H = SHA3-256`, with `hashSize` at least 128 bit.
+1. `KDF = SHAKE256` and `H = SHA3-512`, with `hashSize` at least 256 bit.
```

Comment:
The inclusion of SHAKE has some aspects that need to be considered.
1. SHAKE is not offered by NIST SP-800 56Cr2, so compliance is kind of less then 100%.
2. SHAKE is already kind of included in the list due to KMAC, since KMAC->cSHAKE->SHAKE.
3. SHAKE is an XOF with the usual truncation issue that Trunc_d(SHAKE128(M, d+e)) is identical to SHAKE128(M, d), see FIPS 202 page 24. So I would certainly invest some time thinking about how to rule out that this XOF property does not have critical side effects. My gut feeling: That would at least mean that we must do hashing, i.e. H(ss_i || ct_i). I do realize that there is the "|| fixedInfo" at the end, but one should be careful here, since this is kind of optional.

Reply:
Does this warrant a big EDNOTE in the doc?

Also, what timezone are you in? /slash/ how early do you get up on a Sunday?

Reply:
An ENDNOTE might be a good idea. I like SHAKE, but securing this through breaking the truncation property could end up mimicing just similar padding stuff as KMAC, e.g. taking some non-empty fixedInfo as mandatory. SHAKE is fine as long as you exactly know what do (this might be the reason why it's not an approved hash function).

I am UTC+1. I get up too early, but I enjoy working on excellent documents:-)